pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/PinotSinkFunction.java

```diff
@@ -96,7 +102,7 @@ public void close()
       throws Exception {
     LOG.info("Closing Pinot Sink");
     try {
-      if (_segmentNumRecord > 0) {
+      if (!((StreamingRuntimeContext) this.getRuntimeContext()).isCheckpointingEnabled()) {
```

Comment:
I think that we should still check with `_segmentNumRecord > 0` ?

Reply:
ah yes, but only if checkpointing is enabled right? Otherwise the segment will flush when the jobs fails

Reply:
I meant that we need to flush with the following check?

`_segmentNumRecord > 0 && !((StreamingRuntimeContext) this.getRuntimeContext()).isCheckpointingEnabled()`